feat: use actionCreator for account actions

ReeceHoffmann opened the pull request:
Example implementation of action creation using createAction.

ReeceHoffmann committed:
feat: use actionCreator for account actions

igboyes commented on src/js/account/actions.js:
```diff
@@ -46,9 +49,12 @@ export const updateAccount = update => ({
+            update
+        }
+    };
 });
```
You can write this without a `return`.

igboyes commented on src/js/account/actions.js:
```diff
@@ -59,10 +65,13 @@ export const updateAccountSettings = update => ({
-    oldPassword,
-    newPassword
+export const changePassword = createAction(CHANGE_ACCOUNT_PASSWORD.REQUESTED, (oldPassword, newPassword) => {
+    return {
```
You can write this without a `return`.

igboyes commented on src/js/account/actions.js:
```diff
@@ -103,10 +110,8 @@ export const clearAPIKey = simpleActionCreator(CLEAR_API_KEY);
-    keyId,
-    permissions
+export const updateAPIKey = createAction(UPDATE_API_KEY.REQUESTED, (keyId, permissions) => {
+    return { payload: { keyId, permissions } };
```
You can write this without a `return`.

igboyes reviewed (changes requested):
This looks good to me and I think we should continue with it. I actually like the data being in a nested `payload` object.

Please deal with this warning from GitHub actions:
> Check warning on line 142 in src/js/errors/reducer.js

You can pare down a lot of anonymous arrow functions in this PR by eliminating `return` statements where possible. I only commented a few.

ReeceHoffmann commented:
> This looks good to me and I think we should continue with it. I actually like the data being in a nested `payload` object.
> 
> Please deal with this warning from GitHub actions:
> 
> > Check warning on line 142 in src/js/errors/reducer.js
> 
> You can pare down a lot of anonymous arrow functions in this PR by eliminating `return` statements where possible. I only commented a few.

Sounds good, I will get to work updating the rest of the action creators. I changed all the arrow functions to function as "one line" arrow functions (automatic return) and fixed the oversight that caused that warning.

igboyes commented:
:tada: This PR is included in version 2.4.0 :tada:

The release is available on [GitHub release](https://github.com/virtool/virtool-ui/releases/tag/2.4.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
